### DISCUSSION TITLE
How can I select multiple link and how to change colors and thickness in one shot ?

### DISCUSSION THREAD

**Discussion by Lijesh-km** [clientIO/joint#1973] [Q&A]
### **I am using the kitchen sink app**
1. I want to select multiple links (standard.links) and I need to change selected links' colors or other properties in one short is it possible?
2. I create a stranded rectangle in the stencil. when I drag and drop the element I need a custom label outside the element how can I do it?

**Comment by jamesgeorgewilliams** [comment]
Yes, it is possible. 

1.  Using the following code, you could add links to your selection when pressing the CTRL/Meta key, and change the stroke color.

```js
paper.on('link:pointerup', function(cellView, evt) {
    // Select an element if CTRL/Meta key is pressed while the element is clicked.
    if ((evt.ctrlKey || evt.metaKey)) {
        selection.collection.add(cellView.model);
    }

    selection.collection.models.forEach((model) => {
        const cell = graph.getCell(model.id);
        if (cell.prop('type') === 'standard.Link') {
            cell.attr('line/stroke', 'tomato');
        }
    });
});
```

2.  Our latest Demo Wednesday example deals with a similar question:

https://codepen.io/jointjs/pen/qByZdKZ
https://github.com/clientIO/joint/discussions/1971

You can certainly take some inspiration from it. 

You can take advantage of `dragStartClone` in the Stencil as seen in the linked demo above.

**Reply by Lijesh-km** [reply]
thanks for your response @jamesgeorgewilliams 
I need to do this with help of inspector services.
consider I have 5 lines in my paper but I only need to change the color for selected two links only. I need to select dynamic colors also

**Reply by Lijesh-km** [reply]
2. I need a rectangle it must have two labels one is outside of the element,

**Reply by kumilingus** [reply]
The "inspector service" is just a different name for the inspector plugin (`ui.Inspector`).
Here's an example of a rectangle with 2 labels: https://jsfiddle.net/kumilingus/97dwxqv3/
Should you have other questions unrelated to this thread, please start a new discussion.

**Reply by Lijesh-km** [reply]
![sample](https://user-images.githubusercontent.com/121955632/210732143-b09de93e-2713-40ae-9e9f-597d95c0f51d.png)
@kumilingus 
here you can see I have three links and I selected two links. is it possible to change the color of the two selected links in one shot with the help of the inspector (ui.Inspector).

**Reply by kumilingus** [reply]
Yes, as shown here: https://codepen.io/jointjs/pen/VwMJGxQ
